core/templates/domain/objects/NumberWithUnitsObjectFactory.ts

```diff
@@ -207,8 +207,8 @@ export class NumberWithUnitsObjectFactory {
         units = rawInput.substr(0, ind).trim();
 
         startsWithCorrectCurrencyUnit = false;
-        for (var i = 0; i < keys.length; i++) {
-          for (var j = 0;
+        for (let i = 0; i < keys.length; i++) {
+          for (let j = 0;
             j < ObjectsDomainConstants.CURRENCY_UNITS[
               keys[i]].front_units.length; j++) {
```

Comment:
```suggestion
          let unitLength = ObjectsDomainConstants.CURRENCY_UNITS[keys[i]];
          for (let j = 0; j < unitLength; j++) {
```

Reply:
Done

Reply:
Please do similar change in other parts of this file.

Reply:
done